tests/components/freebox/test_config_flow.py

```diff
@@ -17,13 +18,13 @@
 
 from tests.common import MockConfigEntry
 
-MOCK_ZEROCONF_DATA = {
-    "host": "192.168.0.254",
-    "port": 80,
-    "hostname": "Freebox-Server.local.",
-    "type": "_fbx-api._tcp.local.",
-    "name": "Freebox Server._fbx-api._tcp.local.",
-    "properties": {
+MOCK_ZEROCONF_DATA = zeroconf.HaServiceInfo(
```

Comment:
Why using `HaServiceInfo` here as the flow uses `DiscoveryInfoType` ?

Or the other way around.

Reply:
`HaServiceInfo` is the best type created by zeroconf. But we cannot use it in the config flow due to mypy inheritance where we have to keep `DiscoveryInfoType`

Reply:
See https://github.com/home-assistant/core/blob/a64cec6da1b3d762f902e738b99c05c74d371b31/homeassistant/config_entries.py#L1371-L1375

Reply:
@bdraco do you think we should change the type in `config_entries`?

Reply:
I think how you have it now is fine